msdk-rawdata/msdk-rawdata-peakinvestigator/src/main/java/io/github/msdk/rawdata/peakinvestigator/PeakInvestigatorMsSpectrum.java

```diff
@@ -0,0 +1,125 @@
+/* 
+ * (C) Copyright 2015-2016 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package io.github.msdk.rawdata.peakinvestigator;
+
+import java.util.TreeMap;
+
+import com.google.common.collect.Range;
+
+import io.github.msdk.datamodel.msspectra.MsSpectrum;
+import io.github.msdk.datamodel.msspectra.MsSpectrumType;
+import io.github.msdk.util.tolerances.MzTolerance;
+
+/**
+ * This class is used provide m/z-dependent mass tolerances from
+ * PeakInvestigator by decorating an existing instance of MsSpectrum.
+ *
+ */
+public class PeakInvestigatorMsSpectrum implements MsSpectrum, MzTolerance {
```

Comment:
What sort of filters are you thinking would be appropriate after PI? SGFilter, MeanFilter, and Resample don't seem appropriate for centroided data. CropFilter seems more appropriate prior to PI (i.e. don't process more scans than necessary) or during PI (it can limit analysis to specific mass range).

I do agree that it makes sense to remove ```MzToleranceProvider``` and add ```MsSpectrum.getMzTolerance()```. How do you want to handle ```null``` in methods that expect a MzTolerance? Fallback onto a MzTolerance specified by an individual MSDKMethod like below?

```
        for (MsScan scan : inputScans) {

            if (canceled)
                return null;

            MzTolerance mzTolerance = mzToleranceProvider.getMzTolerance(scan);
            if (mzTolerance == null) {
                 mzTolerance = this.mzTolerance;
            }

            massConnector.addScan(inputFile, scan, mzTolerance);
            processedScans++;
        }
```